File: 项目文件/backend/app/schemas/file.py
"""文件 Pydantic 模型"""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class FileResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    stored_path: str
    size: int
    sha256: str
    mime_type: str
    folder_id: uuid.UUID | None = None
    owner_id: uuid.UUID
    visibility: str
    restricted_users: list[str] | None = None
    restricted_tags: list[str] | None = None
    expires_at: datetime | None = None
    created_at: datetime


class FileListResponse(BaseModel):
    items: list[FileResponse]
    total: int


class FolderResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    parent_id: uuid.UUID | None = None
    owner_id: uuid.UUID
    visibility: str = "private"
    restricted_users: list[str] | None = None
    restricted_tags: list[str] | None = None
    expires_at: datetime | None = None
    unified_management: bool = False
    created_at: datetime


class FolderCreateRequest(BaseModel):
    name: str
    parent_id: uuid.UUID | None = None


class FileUpdateRequest(BaseModel):
    """文件更新请求"""
    expires_at: datetime | None = None


class FolderUpdateRequest(BaseModel):
    """文件夹更新请求"""
    name: str | None = None
    visibility: str | None = None
    restricted_users: list[str] | None = None
    restricted_tags: list[str] | None = None
    expires_at: datetime | None = None
    unified_management: bool | None = None

File: 项目文件/frontend/src/types/file.ts
import type { Visibility } from '../utils/visibility';

export interface FileRecord {
  id: string;
  name: string;
  stored_path: string;
  size: number;
  sha256: string;
  mime_type: string;
  folder_id: string | null;
  owner_id: string;
  visibility: Visibility;
  restricted_users: string[] | null;
  restricted_tags: string[] | null;
  expires_at: string | null;
  created_at: string;
}

export interface Folder {
  id: string;
  name: string;
  parent_id: string | null;
  owner_id: string;
  visibility: Visibility;
  restricted_users: string[] | null;
  restricted_tags: string[] | null;
  expires_at: string | null;
  unified_management: boolean;
  created_at: string;
}

export interface FileListResponse {
  items: FileRecord[];
  total: number;
}

export interface FolderCreateRequest {
  name: string;
  parent_id?: string;
}

export interface FolderUpdateRequest {
  name?: string;
  visibility?: Visibility;
  restricted_users?: string[];
  restricted_tags?: string[];
  expires_at?: string | null;
  unified_management?: boolean;
}
